openstudiocore/src/model/GeneratorMicroTurbine.cpp

```diff
@@ -0,0 +1,1004 @@
+      {
+        LOG(briefDescription() << " does not belong to the same model as the curve you want to set.");
+        return false;
+      }
+      result = setPointer(OS_Generator_MicroTurbineFields::ExhaustAirTemperatureFunctionofTemperatureCurveName, curve.get().handle());
+    }
+    else {
+      resetExhaustAirTemperatureFunctionofTemperatureCurve();
+      result = true;
+    }
+    return result;
+  }
+
+  void GeneratorMicroTurbine_Impl::resetExhaustAirTemperatureFunctionofTemperatureCurve() {
+    bool result = setString(OS_Generator_MicroTurbineFields::ExhaustAirTemperatureFunctionofTemperatureCurveName, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorMicroTurbine_Impl::setExhaustAirTemperatureFunctionofPartLoadRatioCurve(const boost::optional<Curve>& curve) {
+    bool result(false);
+    if (curve) {
+      if(model() != curve.get().model())
+      {
+        LOG(briefDescription() << " does not belong to the same model as the curve you want to set.");
+        return false;
+      }
+      result = setPointer(OS_Generator_MicroTurbineFields::ExhaustAirTemperatureFunctionofPartLoadRatioCurveName, curve.get().handle());
+    }
+    else {
+      resetExhaustAirTemperatureFunctionofPartLoadRatioCurve();
+      result = true;
+    }
+    return result;
+  }
+
+  void GeneratorMicroTurbine_Impl::resetExhaustAirTemperatureFunctionofPartLoadRatioCurve() {
+    bool result = setString(OS_Generator_MicroTurbineFields::ExhaustAirTemperatureFunctionofPartLoadRatioCurveName, "");
+    OS_ASSERT(result);
+  }
+
+  boost::optional<Curve> GeneratorMicroTurbine_Impl::optionalElectricalPowerFunctionofTemperatureandElevationCurve() const {
+    return getObject<ModelObject>().getModelObjectTarget<Curve>(OS_Generator_MicroTurbineFields::ElectricalPowerFunctionofTemperatureandElevationCurveName);
+  }
+
+  boost::optional<Curve> GeneratorMicroTurbine_Impl::optionalElectricalEfficiencyFunctionofTemperatureCurve() const {
+    return getObject<ModelObject>().getModelObjectTarget<Curve>(OS_Generator_MicroTurbineFields::ElectricalEfficiencyFunctionofTemperatureCurveName);
+  }
+
+  boost::optional<Curve> GeneratorMicroTurbine_Impl::optionalElectricalEfficiencyFunctionofPartLoadRatioCurve() const {
+    return getObject<ModelObject>().getModelObjectTarget<Curve>(OS_Generator_MicroTurbineFields::ElectricalEfficiencyFunctionofPartLoadRatioCurveName);
+  }
+
+} // detail
+
+GeneratorMicroTurbine::GeneratorMicroTurbine(const Model& model)
+  : Generator(GeneratorMicroTurbine::iddObjectType(),model)
+{
+  OS_ASSERT(getImpl<detail::GeneratorMicroTurbine_Impl>());
+
+  // TODO: Appropriately handle the following required object-list fields.
+  //     OS_Generator_MicroTurbineFields::ElectricalPowerFunctionofTemperatureandElevationCurveName
+  //     OS_Generator_MicroTurbineFields::ElectricalEfficiencyFunctionofTemperatureCurveName
+  //     OS_Generator_MicroTurbineFields::ElectricalEfficiencyFunctionofPartLoadRatioCurveName
+  bool ok = true;
```

Comment:
Where will you set the fields that are required but have no default?  @kbenne @mbadams5 can advise

referenceElectricalPowerOutput
referenceElectricalEfficiencyUsingLowerHeatingValue
electricalPowerFunctionofTemperatureandElevationCurve
electricalEfficiencyFunctionofTemperatureCurve
electricalEfficiencyFunctionofPartLoadRatioCurve